io/io/src/TStreamerInfo.cxx

```diff
@@ -262,6 +262,7 @@ namespace {
       Int_t datasize = 1;
       auto memClass = TClass::GetClass(localtypename.c_str());
       std::vector<Int_t> dimensions;
+      static TClassRef string_classref("string");
```

Comment:
Can this be
```suggestion
      static const TClassRef string_classref("string");
```
?

Reply:
In practice maybe.  In theory no.  The point of the `TClassRef` is that it is first initialized with a `const char *` and then updated to point to the `TClass` if it was not yet available at the time of construction.

Reply:
Is the update a thread-safe operation, or is there something external that guarantees the global variable is not mutated concurrently?

Reply:
It is meant to be thread-safe (re-reviewing the code now).

Reply:
The underlying data is a pointer to an atomic pointer.   A new commit now make the outer pointer also atomic to avoid any risk of partial reads.